Userland/Services/DeviceMapper/DeviceEventLoop.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "DeviceNodeFamily.h"
+#include <AK/ByteBuffer.h>
+#include <Kernel/API/DeviceEvent.h>
+#include <Kernel/API/DeviceFileTypes.h>
+#include <LibCore/EventLoop.h>
+#include <LibCore/File.h>
+#include <LibCore/Notifier.h>
+#include <LibIPC/MultiServer.h>
+
+namespace DeviceMapper {
+
+class DeviceEventLoop {
+public:
+    enum class MinorNumberAllocationType {
+        SequentialUnlimited,
+        SequentialLimited,
+    };
+
+    struct device_node_match {
+        StringView permission_group;
+        StringView family_type_literal;
+        StringView path_pattern;
+        bool block_device;
+        MajorNumber major_number;
+        MinorNumberAllocationType minor_number_allocation_type;
+        MinorNumber minor_number_start;
+        size_t minor_number_range_size;
+        mode_t create_mode;
+    };
+
+    DeviceEventLoop(NonnullOwnPtr<Core::File>);
+    virtual ~DeviceEventLoop() = default;
+
+    void drain_events_from_devctl();
+
+private:
+    DeviceNodeFamily* find_device_node_family(bool block_device, MajorNumber major_number, MinorNumber minor_number) const;
+    ErrorOr<DeviceNodeFamily*> find_or_register_new_device_node_family(device_node_match const& match, bool block_device, MajorNumber major_number, MinorNumber minor_number);
```

Comment:
Since we cannot have `ErrorOr<DeviceNodeFamily&>` I decided that `DeviceNodeFamily` should be Ref-Counted.
Really this makes sense, because in the future when other servers and services will query the `DeviceMapper` service, we will need to iterate over the list of `DeviceNodeFamily`s, and we definitely don't want to get into a situation where such object is removed while we use it.